lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -0,0 +1,1236 @@
+        result, changed = call_method(client, module, 'stop_db_instance', parameters)
+    elif state == 'started' and instance['DBInstanceStatus'] not in ['available', 'starting', 'restarting']:
+        result, changed = call_method(client, module, 'start_db_instance', parameters)
+    return changed
+
+
+def main():
+    arg_spec = dict(
+        state=dict(choices=['present', 'absent', 'running', 'started', 'stopped', 'rebooted', 'restarted'], default='present'),
+        creation_source=dict(choices=['snapshot', 's3', 'instance']),
+        force_update_password=dict(type='bool', default=False),
+        promote=dict(type='bool', default=False),
+        purge_cloudwatch_logs_exports=dict(type='bool', default=True),
+        purge_tags=dict(type='bool', default=True),
+        wait=dict(type='bool', default=True),
+    )
+
+    parameter_options = dict(
+        allocated_storage=dict(type='int'),
+        allow_major_version_upgrade=dict(type='bool'),
+        apply_immediately=dict(type='bool', default=False),
+        auto_minor_version_upgrade=dict(type='bool'),
+        availability_zones=dict(type='list', aliases=['zones', 'az']),
+        backup_retention_period=dict(type='int'),
+        ca_certificate_identifier=dict(),
+        character_set_name=dict(),
+        copy_tags_to_snapshot=dict(type='bool'),
+        db_cluster_identifier=dict(aliases=['cluster_id']),
+        db_instance_class=dict(aliases=['class']),
+        db_instance_identifier=dict(required=True, aliases=['instance_id', 'id']),
+        db_name=dict(),
+        db_parameter_group_name=dict(),
+        db_security_groups=dict(),
+        db_snapshot_identifier=dict(),
+        db_subnet_group_name=dict(),
+        domain=dict(),
+        domain_iam_role_name=dict(),
+        enable_cloudwatch_logs_exports=dict(type='list'),
+        enable_iam_database_authentication=dict(type='bool'),
+        enable_performance_insights=dict(type='bool'),
+        engine=dict(choices=[
+            "aurora",
+            "aurora-mysql",
+            "aurora-postgresql",
+            "mariadb",
+            "mysql",
+            "oracle-ee",
+            "oracle-se",
+            "oracle-se1",
+            "oracle-se2",
+            "postgres",
+            "sqlserver-ee",
+            "sqlserver-ex",
+            "sqlserver-se",
+            "sqlserver-web",
+        ]),
+        engine_version=dict(),
+        final_db_snapshot_identifier=dict(aliases=['final_snapshot_identifier']),
+        force_failover=dict(type='bool'),
+        iops=dict(type='int'),
+        kms_key_id=dict(),
+        license_model=dict(choices=['license-included', 'bring-your-own-license', 'general-public-license']),
+        master_user_password=dict(aliases=['password'], no_log=True),
+        master_username=dict(aliases=['username']),
+        monitoring_interval=dict(),
+        monitoring_role_arn=dict(),
+        multi_az=dict(type='bool'),
```

Comment:
Looking at the docs (https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBInstance.html) I think multi_az should be mutually exclusive with zones. ``Specifies if the DB instance is a Multi-AZ deployment. You can't set the AvailabilityZone parameter if the MultiAZ parameter is set to true. ``

Reply:
Right, setting multi_az to true doesn't seem to be doing anything special currently (my RDS postgre instance ends up being created without multi-az enabled)